computer_vision/cnn_classification/cnn_classification.py

```diff
@@ -0,0 +1,98 @@
+"""
+Convolutional Neural Network
+
+Objective : To train a CNN model detect if TB is present in Lung X-ray or not.
+
+Resources CNN Theory :
+    https://en.wikipedia.org/wiki/Convolutional_neural_network
+Resources Tensorflow : https://www.tensorflow.org/tutorials/images/cnn
+
+Download dataset from :
+https://lhncbc.nlm.nih.gov/LHC-publications/pubs/TuberculosisChestXrayImageDataSets.html
+
+1. Download the dataset folder and create two folder training set and test set
+in the parent dataste folder
+2. Move 30-40 image from both TB positive and TB Negative folder
+in the test set folder
+3. The labels of the iamges will be extracted from the folder name
+the image is present in.
+
+"""
+
+# Part 1 - Building the CNN
+
+import numpy as np
+
+# Importing the Keras libraries and packages
+import tensorflow as tf
+from tensorflow.keras import layers, models
+
+if __name__ == "__main__":
+
+    # Initialising the CNN
+    classifier = models.Sequential()
+
+    # Step 1 - Convolution
+    classifier.add(
+        layers.Conv2D(32, (3, 3), input_shape=(64, 64, 3), activation="relu")
+    )
+
+    # Step 2 - Pooling
+    classifier.add(layers.MaxPooling2D(pool_size=(2, 2)))
+
+    # Adding a second convolutional layer
+    classifier.add(layers.Conv2D(32, (3, 3), activation="relu"))
+    classifier.add(layers.MaxPooling2D(pool_size=(2, 2)))
+
+    # Step 3 - Flattening
+    classifier.add(layers.Flatten())
+
+    # Step 4 - Full connection
+    classifier.add(layers.Dense(units=128, activation="relu"))
+    classifier.add(layers.Dense(units=1, activation="sigmoid"))
+
+    # Compiling the CNN
+    classifier.compile(
+        optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"]
+    )
+
+    # Part 2 - Fitting the CNN to the images
+
+    # Load Trained model weights
+
+    # from keras.models import load_model
+    # regressor=load_model('cnn.h5')
```

Comment:
Is this required? If not, please remove them.

Reply:
yes in case someone wants to reuse trained model

Reply:
request changes done.

Reply:
requested changes done